readium/navigator/src/main/java/org/readium/r2/navigator/media3/tts/TtsEngineFacade.kt

```diff
@@ -45,54 +43,44 @@ internal class TtsEngineFacade<S : TtsEngine.Settings, P : TtsEngine.Preferences
         engine.close()
     }
 
+    private var currentTask: UtteranceTask<E>? = null
+
     private data class UtteranceTask<E : TtsEngine.Error>(
         val requestId: TtsEngine.RequestId,
         val continuation: CancellableContinuation<E?>,
         val onRange: (IntRange) -> Unit
     )
 
+    private fun getTask(id: TtsEngine.RequestId) =
+        currentTask?.takeIf { it.requestId == id }
+
+    private fun popTask(id: TtsEngine.RequestId) =
+        getTask(id)
+            ?.also { currentTask = null }
```

Comment:
Is that the important thing? I can't see any difference in the listener, except this. Which call to `currentTask` happened to receive the wrong task?

Reply:
Yes. It was `onFlushed`, which was called with the ID of the previous task than the `currentTask` one.

As we didn't check the id before setting `currentTask = null`, it was breaking the playback.